Ical.Net/CalendarCollection.cs

```diff
@@ -39,85 +39,57 @@ public static CalendarCollection Load(TextReader tr)
         return collection;
     }
 
-    public HashSet<Occurrence> GetOccurrences(IDateTime dt)
+    public HashSet<Occurrence> GetOccurrencesOfDay(IDateTime dt)
     {
         var occurrences = new HashSet<Occurrence>();
         foreach (var iCal in this)
         {
-            occurrences.UnionWith(iCal.GetOccurrences(dt));
+            occurrences.UnionWith(iCal.GetOccurrencesOfDay(dt));
         }
         return occurrences;
     }
 
-    public HashSet<Occurrence> GetOccurrences(DateTime dt)
-    {
-        var occurrences = new HashSet<Occurrence>();
-        foreach (var iCal in this)
-        {
-            occurrences.UnionWith(iCal.GetOccurrences(dt));
-        }
-        return occurrences;
-    }
-
-    public HashSet<Occurrence> GetOccurrences(IDateTime startTime, IDateTime endTime)
-    {
-        var occurrences = new HashSet<Occurrence>();
-        foreach (var iCal in this)
-        {
-            occurrences.UnionWith(iCal.GetOccurrences(startTime, endTime));
-        }
-        return occurrences;
-    }
-
-    public HashSet<Occurrence> GetOccurrences(DateTime startTime, DateTime endTime)
-    {
-        var occurrences = new HashSet<Occurrence>();
-        foreach (var iCal in this)
-        {
-            occurrences.UnionWith(iCal.GetOccurrences(startTime, endTime));
-        }
-        return occurrences;
-    }
-
-    public HashSet<Occurrence> GetOccurrences<T>(IDateTime dt) where T : IRecurringComponent
-    {
-        var occurrences = new HashSet<Occurrence>();
-        foreach (var iCal in this)
-        {
-            occurrences.UnionWith(iCal.GetOccurrences<T>(dt));
-        }
-        return occurrences;
-    }
-
-    public HashSet<Occurrence> GetOccurrences<T>(DateTime dt) where T : IRecurringComponent
-    {
-        var occurrences = new HashSet<Occurrence>();
-        foreach (var iCal in this)
-        {
-            occurrences.UnionWith(iCal.GetOccurrences<T>(dt));
-        }
-        return occurrences;
-    }
-
-    public HashSet<Occurrence> GetOccurrences<T>(IDateTime startTime, IDateTime endTime) where T : IRecurringComponent
-    {
-        var occurrences = new HashSet<Occurrence>();
-        foreach (var iCal in this)
-        {
-            occurrences.UnionWith(iCal.GetOccurrences<T>(startTime, endTime));
-        }
-        return occurrences;
-    }
-
-    public HashSet<Occurrence> GetOccurrences<T>(DateTime startTime, DateTime endTime) where T : IRecurringComponent
-    {
-        var occurrences = new HashSet<Occurrence>();
-        foreach (var iCal in this)
-        {
-            occurrences.UnionWith(iCal.GetOccurrences<T>(startTime, endTime));
-        }
-        return occurrences;
-    }
+    public IEnumerable<Occurrence> GetOccurrencesOfDay(DateTime dt)
```

Comment:
Would you like to keep the unused (and thus uncovered overloads)?

Reply:
I generally didn't remove any methods, because I don't want to change the API in this PR outside of what is related to the immediate topic. If we want to change it, I think we should do that in a separate PR, so we keep concerns separated. But this one indeed is a special case. Actually I don't think, we should have a separate method just for getting the occurrences for a single day, because it can easily be implemented with the other variants and its not obvious, why we wouldn't also add versions for ByWeek, ByHour, etc. So I'd certainly vote for removing them, but also the other overloads (i.e. `GetOccurrencesOfDay(IDateTime)`).

In this case we have another challenge. Due to the params of `GetOccurrences(DateTime? = null, DateTime? = null)` et al being optional now, a call like `GetOccurrences(new DateTime(...))` would now invoke the overload with startTime, endTime, rather than the overload for getting the occurrences for a single day (which has been renamed). This could break existing code without being obvious. It will be introduced in v5, so we could argue that's ok (and I actually would do so), but we should be aware of it. And add a note to a potential migration guide. What do you think?

Reply:
@axunonb I removed the versions of `GetOccurrences(date)` that were handling a single day altogether (see prev comment). Only two tests existed that tested any of them at all. I don't think, they will be missing. We should add some comment in a migration guide, should one be written, about the ambiguity due to the new nullable params in the other overloads.